cinder/tests/unit/scheduler/test_shard_filter.py

```diff
@@ -135,3 +135,18 @@ def test_shard_override_no_data(self):
         host = fakes.FakeBackendState('host1', {'capabilities': caps})
         self.props['scheduler_hints'] = {'vcenter-shard': None}
         self.assertFalse(self.filt_cls.backend_passes(host, self.props))
+
+    def test_sharding_enabled_any_backend_match(self):
+        self.filt_cls._PROJECT_SHARD_CACHE['baz'] = ['sharding_enabled']
+        self.props['request_spec']['volume_properties']['project_id'] = 'baz'
+        caps = {'vcenter-shard': 'vc-a-0'}
+        host = fakes.FakeBackendState('host1', {'capabilities': caps})
+        self.assertTrue(self.filt_cls.backend_passes(host, self.props))
+
+    def test_sharding_enabled_and_single_shard_any_backend_match(self):
+        self.filt_cls._PROJECT_SHARD_CACHE['baz'] = ['sharding_enabled',
+                                                     'vc-a-1']
+        self.props['request_spec']['volume_properties']['project_id'] = 'baz'
+        caps = {'vcenter-shard': 'vc-a-0'}
+        host = fakes.FakeBackendState('host1', {'capabilities': caps})
+        self.assertTrue(self.filt_cls.backend_passes(host, self.props))
```

Comment:
Should we add a test for the scheduler-hint overwriting `sharding_enabled`? Because it looks like it would do that and if we have a test for it, then we acknowledge that this is wanted behavior.

Reply:
Yes. It's done.